### ISSUE TITLE
[BUG] CF Block not working properly?

### ISSUE THREAD

**Issue by FaceSteel** [openbullet/openbullet#726]
**Describe the bug**
Whenever you use HTTP proxies while executing a CF block, it returns an error. 
"CF Bypass Failed: Cloudflare [JavaScript]: Something wrong happened"

1. Go to your config
2. Create a CF block 
3. Use a valid HTTP proxy
4. Hit "start"
5. See error

**Expected behavior**
It's expected that the block should bypass CF clearance without problem, it does so WITHOUT proxies, but doesn't matter how quality your proxies are, it'll fail bypassing cloudflare

**Screenshots**
https://i.imgur.com/fFS61qY.png

**Version**
Latest

**Comment by FaceSteel** [comment]
And sometimes, it'll just say "ERROR: A task was canceled."

**Comment by openbullet** [comment]
CF updated their protection and it's much harder now, I don't know if and when the new bypass will be implemented

**Comment by FaceSteel** [comment]
> CF updated their protection and it's much harder now, I don't know if and when the new bypass will be implemented

Right now the only workaround I've found to this issue is using proxifier with a proxy rotating on each request. It's the only way.

**Comment by Artarex** [comment]
Maybe in OB2 you can implement pupeteer. 
CPU usage should be much lower compared to selenium when it works headless. 
And with pupeteer CF shouldn't be a problem since it loads / gets all cookies / js etc

**Comment by openbullet** [comment]
I was not using selenium to solve cloudflare anyways, I was using CloudflareSolverRe which uses jint directly to run the js.

**Comment by openbullet** [comment]
I will look into pupeteer as a permanent selenium replacement though

**Comment by openbullet** [comment]
It doesn't seem to support proxies though, or at least its .net port. This might be a problem.

**Comment by Artarex** [comment]
> 
> 
> I was not using selenium to solve cloudflare anyways, I was using CloudflareSolverRe which uses jint directly to run the js.

Yep, that was the reason why i suggested pupeeteer. Because the new cloudflare protection can NOT really be solved anymore with CloudFlareSolverRe because its more than just javascript based now. They also store cookies, do sometimes multiple redirects etc. 

So the only solution for now would be using selenium and "act" like a normal user. Howevery selenium is of course using a lot of cpu if you want multiple threads. And pupeeteer should be much better because it is headless. 


Hope you find a workaround for the proxy problem :)

**Comment by openbullet** [comment]
@Artarex selenium can work with headless browsers as well though. I'll have to see if there is an actual performance increase or not.

**Comment by h0ttz** [comment]
Hi @openbullet Any update on this ?

**Comment by openbullet** [comment]
@h0ttz I have exams at the moment so I cannot code for the next 3 weeks, and even after that I doubt I will be able to implement a solution as cloudflare completely reworked their system. I will have to either use puppeteer / selenium or to find a bypass in py/node and port it.

**Comment by drysauce** [comment]
@openbullet its been a while, whats happening with CF now?

**Comment by openbullet** [comment]
@drysauce This project is discontinued. I am looking into possible solutions for this using FlareSolverrSharp on OB2 but had little luck so far.
